torch/_torch_docs.py

```diff
@@ -2698,6 +2707,8 @@
 
 If :attr:`mat` is a `n x m` Tensor, :attr:`vec` is a 1D Tensor of size `m`, :attr:`out` will be 1D of size `n`.
 
+.. note:: This function does not :ref:`broadcast <broadcasting-semantics>`.
```

Comment:
Good questions.

There are 5 relevant functions: mm, bmm, mv, ger, dot.

In the way I've structured the documentation, I only mention matmul in cases where there is broadcast behavior that allows you to do the equivalent ops with broadcasting; So: bmm (broadcast over batches) and mm (really squeeze/unsqueeze if one of the ops is 1-d, but close enough).

There is equivalent behavior in matmul independent of broadcasting in bmm, mm, dot, mv.  There isn't really in ger (you have to manually unsqueeze/squeeze the elements to get the equivalent).

So, there are really 3 combinations:
1) equivalent + broadcasting: mm, bmm
2) equivalent: dot, mv
3) not equivalent: ger

I prefer the term "more general" for matmul over "low level" for (mm, bmm, mv, dot).  So maybe the right thing to do for these cases are:
1) Mention "more general" matmul with broadcasting support
2) Mention "more general" matmul (no broadcasting support is separate note)
3) Leave as is

Thoughts?